src/resource_api/schema.py

```diff
@@ -190,8 +190,21 @@ def get_schema(self):
 
 
 class IndexableField(BaseSimpleField):
+    """
+    Base class for most of the primitive fields.
+
+    choices (list|tuple = None)
+        Predefined list of possible values for the field. ValidationError will be raised if field value doesn't
+        match any of the defined values.
+    choice_labels (list|tuple = None)
```

Comment:
However, for the primitive cases it would also make sense to have just a list of choices that would internally be transformed from [val1, val2, val3] into {val1:val1, val2:val2, val3:val3}